### PR TITLE
Addressing `yarn run dist` issues.

### PR DESCRIPTION
**Pull request by nitsakh**
I see two issues running `yarn run dist`, on latest master.

1. Problem with `node-gyp`

```
yarn run v0.24.6
$ npm run build && cross-env BABEL_ENV=production babel --out-file app/dist/bundle.js --no-comments --minified app/dist/bundle.js && build 

> @ build /Users/nitish/Dev/myProjects/hyper
> cross-env NODE_ENV=production webpack

Hash: 05f8d3410eda03759871
Version: webpack 2.6.1
Time: 4702ms
           Asset     Size  Chunks                    Chunk Names
       bundle.js  1.64 MB       0  [emitted]  [big]  main
   bundle.js.map  2.04 MB       0  [emitted]         main
assets/icons.svg  3.34 kB          [emitted]         
   [5] ./lib/utils/plugins.js 9.52 kB {0} [built]
   [6] ./~/react/react.js 56 bytes {0} [built]
  [14] ./lib/rpc.js 57 bytes {0} [built]
  [29] ./lib/actions/sessions.js 3.23 kB {0} [built]
  [30] ./lib/actions/term-groups.js 4.08 kB {0} [built]
  [40] ./~/react-redux/es/index.js 230 bytes {0} [built]
  [67] ./lib/actions/notifications.js 347 bytes {0} [built]
  [70] ./~/react-dom/index.js 59 bytes {0} [built]
 [128] ./lib/actions/config.js 245 bytes {0} [built]
 [129] ./lib/actions/index.js 207 bytes {0} [built]
 [130] ./lib/containers/hyper.js 3.97 kB {0} [built]
 [131] ./lib/store/configure-store.js 273 bytes {0} [built]
 [132] ./lib/utils/config.js 346 bytes {0} [built]
 [133] ./~/react-deep-force-update/lib/index.js 1.45 kB {0} [built]
 [154] ./lib/index.js 3.98 kB {0} [built]
    + 300 hidden modules
electron-builder 18.3.5
Rebuilding native production dependencies for darwin:x64
Rebuilding native dependency node-pty
Error: /usr/local/Cellar/node/7.10.0/bin/node exited with code 1
Output:
yarn run v0.24.6
$ node scripts/install.js 
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

Error output:
events.js:163
      throw er; // Unhandled 'error' event
      ^

Error: spawn node-gyp ENOENT
    at exports._errnoException (util.js:1050:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:193:32)
    at onErrorNT (internal/child_process.js:367:16)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)
    at Module.runMain (module.js:607:11)
    at run (bootstrap_node.js:427:7)
    at startup (bootstrap_node.js:151:9)
    at bootstrap_node.js:542:3
error Command failed with exit code 1.

    at exports._errnoException (util.js:1050:11)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:193:32)
    at onErrorNT (internal/child_process.js:367:16)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)
    at Module.runMain (module.js:607:11)
    at run (bootstrap_node.js:427:7)
    at startup (bootstrap_node.js:151:9)
    at bootstrap_node.js:542:3
    at ChildProcess.childProcess.once.code (..../hyper/node_modules/electron-builder-util/src/util.ts:135:14)
    at Object.onceWrapper (events.js:293:19)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:194:7)
    at maybeClose (internal/child_process.js:899:16)
    at Socket.<anonymous> (internal/child_process.js:342:11)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:191:7)
    at Pipe._handle.close [as _onclose] (net.js:511:12)
```

2. macOS code sign issue.

I have a PR which tries to address these. Please comment, and suggest.

### CODE REVIEW

**Comment by albinekb** [comment]
Nice!

File: yarn.lock [hunk 1]
```diff
@@ -4069,7 +4069,7 @@ node-forge@^0.7.1:
   version "0.7.1"
   resolved "https://registry.yarnpkg.com/node-forge/-/node-forge-0.7.1.tgz#9da611ea08982f4b94206b3beb4cc9665f20c300"
 
-node-gyp@^3.6.0:
+node-gyp@^3.6.0, node-gyp@^3.6.2:
```
Comment (chabou) [inline yarn.lock]: It is outdated by @albinekb updates. But this is not a thing

File: package.json [hunk 2]
```diff
@@ -151,6 +151,7 @@
     "eslint-config-xo-react": "0.10.0",
     "eslint-plugin-react": "6.7.1",
     "husky": "0.13.4",
+     "node-gyp": "3.6.2",
```
Comment (Stanzilla) [inline package.json]: one space too much

File: readme.md [hunk 3]
```diff
@@ -52,6 +52,11 @@ make sure its build process is working correctly by running `yarn run rebuild-no
 If you're on macOS, this typically is related to Xcode issues (like not having agreed
 to the Terms of Service by running `sudo xcodebuild` after a fresh Xcode installation).
 
+### Error with `codesign` on mac when running `yarn run dist`
```
Comment (Stanzilla) [inline readme.md]: *macOS

File: readme.md [hunk 4]
```diff
@@ -52,6 +52,11 @@ make sure its build process is working correctly by running `yarn run rebuild-no
...
+
+If you have issues in the `codesign` step when running `yarn run dist` on mac, you can temporarily disable code signing locally by setting
```
Comment (Stanzilla) [inline readme.md]: same as above

**Comment by nitsakh** [comment]
Thanks folks for the review! 👍🏻 I updated the branch based on comments.

**Comment by chabou** [comment]
@nitsakh Great! Can you commit your updated `yarn.lock` too?

**Comment by nitsakh** [comment]
@chabou Sorry! Hadn't noticed that! Done now.

**Comment by albinekb** [comment]
Thanks @nitsakh 🙌 💛

**Comment by chabou** [comment]
Awesome @nitsakh 🎉

**Pull request merged**